nick_work/fdaaa_port.py

```diff
@@ -0,0 +1,372 @@
+
+            #fda_reg_drug
+            if soup.is_fda_regulated_drug is None:
+                fda_reg_drug = None
+            else:
+                fda_reg_drug = (soup.is_fda_regulated_drug).text
+
+            #fda_reg_device
+            if soup.is_fda_regulated_device is None:
+                fda_reg_device = None
+            else:
+                fda_reg_device = (soup.is_fda_regulated_device).text
+
+            #primary_purpose
+            if soup.find('primary_purpose') is not None:
+                primary_purpose = soup.find('primary_purpose').text
+            else:
+                primary_purpose = None
+
+            #is_fda_regulated
+            df_row = fda_reg_df.loc[fda_reg_df['nct_id'] == nct_id]
+            if df_row.empty:
+                is_fda_regulated = None
+            else:
+                is_fda_regulated = df_row['is_fda_regulated'].values[0]
+
+            #study_status
+            study_status = (soup.overall_status).text
+
+            #start_date
+            sd = soup.start_date
+            if sd is not None:
+                try:
+                    start_date = datetime.strptime(sd.text, '%B %d, %Y').date()
+                except ValueError:
+                    start_date = datetime.strptime(sd.text, '%B %Y').date() + relativedelta(months=+1) - timedelta(days=1)
+            else:
+                start_date = None
+
+            #available_completion_date
+            pcd = soup.primary_completion_date
+            if pcd is not None:
+                try:
+                    primary_completion_date = datetime.strptime(pcd.text, '%B %d, %Y').date()
+                    defaulted_pcd = 0
+                except ValueError:
+                    primary_completion_date = datetime.strptime(pcd.text, '%B %Y').date() + relativedelta(months=+1) - timedelta(days=1)
+                    defaulted_pcd = 1
+            else:
+                primary_completion_date = None
+                defaulted_pcd = 0
+
+            cd = soup.completion_date
+            if cd is not None:
+                try:
+                    completion_date = datetime.strptime(cd.text, '%B %d, %Y').date()
+                    defaulted_cd = 0
+                except ValueError:
+                    completion_date = datetime.strptime(cd.text, '%B %Y').date() + relativedelta(months=+1) - timedelta(days=1)
+                    defaulted_cd = 1
+            else:
+                completion_date = None
+                defaulted_cd = 0
+
+            if primary_completion_date == None and completion_date == None:
+                available_completion_date = None
+            elif primary_completion_date == None and completion_date is not None:
+                available_completion_date = completion_date
+                used_cd = 1
+                used_pcd = 0
+            else:
+                available_completion_date = primary_completion_date
+                used_pcd = 1
+                used_cd = 0
+
```

Comment:
This would be more readable:

```python
            if not primary_completion_date and not completion_date:
                available_completion_date = None
            elif completion date and not primary_completion_date:
                available_completion_date = completion_date
                used_cd = True
                used_pcd = False
```

Reply:
Also: you rename `used_pcd` to `used_primary_completion_date` later. You might as well just call it `used_primary_completion_date` and save a line of code (same goes for `used_cd`)

Reply:
Done.